src/models/Function/FunctionService.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import WebSiteManagementClient from "azure-arm-website";
+import { ServiceClientCredentials, WebResource } from "ms-rest";
+import * as request from 'request-promise';
+import { appendExtensionUserAgent, createAzureClient } from "vscode-azureextensionui";
+import * as Constants from "../../constants";
+import { ApisTreeItem } from "../../explorer/ApisTreeItem";
+import { ApiTreeItem } from "../../explorer/ApiTreeItem";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../localize";
+import { nRequest, requestUtil, sendRequest } from "../../utils/requestUtil";
+import { signRequest } from "../../utils/signRequest";
+import { IAzureClientInfo } from "../Azure/AzureClientInfo";
+import { IFunctionKeys } from "./Function";
+import { IFunctionContract } from "./IFunctionContract";
+
+export namespace FunctionService {
```

Comment:
Create a class instead and inherit from a baseclass called ARMService

Please name this class WebAppService or FunctionAppService

Have this logic in the baseclass

// Create client for webiste mgmt client
    export function getClient(node: ApisTreeItem | ApiTreeItem): WebSiteManagementClient {
        const clientInfo: IAzureClientInfo = {
            credentials: node.root.credentials,
            subscriptionId: node.root.subscriptionId,
            environment: node.root.environment
        };
        return createAzureClient(clientInfo, WebSiteManagementClient);
    }

I plan to use this baseclass as well when implementing for gateways.

Reply:
I didn't find ARMservice within this repo, or other repos, do you mean I should create one?

Reply:
yes create a new one